File: bsp4rs/src/com/jetbrains/bsp/bsp4rs/jvm/jvm_environment_item.rs
use serde::{Deserialize, Serialize};

use crate::*;
use std::collections::BTreeMap;

#[derive(Clone, Debug, Default, Eq, PartialEq, Hash, Ord, PartialOrd, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct JvmEnvironmentItem {
    #[serde(default)]
    pub target: BuildTargetIdentifier,
    #[serde(default)]
    pub classpath: Vec<String>,
    #[serde(default)]
    pub jvm_options: Vec<String>,
    #[serde(default)]
    pub working_directory: String,
    #[serde(default)]
    pub environment_variables: BTreeMap<String, String>,
    #[serde(default, skip_serializing_if = "Vec::is_empty")]
    pub main_classes: Vec<JvmMainClass>,
}

File: bsp4kt/src/org/jetbrains/bsp/bsp4kt/JvmEnvironmentItem.kt
package org.jetbrains.bsp.bsp4kt

import kotlinx.serialization.Serializable
import kotlinx.serialization.SerialName
import kotlinx.serialization.json.JsonElement

@Serializable
data class JvmEnvironmentItem(
  val target: BuildTargetIdentifier,
  val classpath: List<String>,
  val jvmOptions: List<String>,
  val workingDirectory: String,
  val environmentVariables: EnvironmentVariables,
  val mainClasses: List<JvmMainClass>? = null
)
